trie/merkle.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+
+import hashlib
+import json
+import sys
+
+## Graypaper conforming implementation
+def hash(data):
+    return hashlib.blake2b(data, digest_size=32).digest()
+
+def fork(l, r):
+    assert len(l) == 32
+    assert len(r) == 32
+    head = l[0] & 0xfe
+    return bytes([head]) + l[1:] + r
+
+def leaf(k, v):
+    if len(v) <= 32:
+        head = 0b01 | (len(v) << 2)
```

Comment:
This is meant to represent `[1, 0] ⌢ bits(E1(∣v∣))`? the first flag bits seem flipped to me.

Reply:
`bits` function puts the least significant bit first. So the octet (byte) that is formed with `[1, 0] ⌢ bits(E1(∣v∣))` has the least significant bit as 1 and the next bit as 0.